test/mktemp.js

```diff
@@ -0,0 +1,112 @@
+assert.equal(result.length, 1);
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isFile());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Directory
+result = shell.mktemp('-d');
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 1);
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isDirectory());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Custom Template
+result = shell.mktemp(path.resolve('.', 'tmp/tmp.XXX'));
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 1);
+assert.equal(result[0].slice(0, -3), path.resolve('.', 'tmp/tmp.'));
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isFile());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Custom Template (Directory)
+result = shell.mktemp('-d', path.resolve('.', 'tmp/tmp.XXX'));
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 1);
+assert.equal(result[0].slice(0, -3), path.resolve('.', 'tmp/tmp.'));
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isDirectory());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Custom Templates
+result = shell.mktemp(path.resolve('.', 'tmp/tmp.AXXX'), path.resolve('.', 'tmp/tmp.BXXX'));
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 2);
+assert.equal(result[0].slice(0, -3), path.resolve('.', 'tmp/tmp.A'));
+assert.equal(result[1].slice(0, -3), path.resolve('.', 'tmp/tmp.B'));
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isFile());
+assert.ok(fs.existsSync(result[1]));
+assert.ok(fs.statSync(result[1]).isFile());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+if (isNotWindows) assert.equal(fs.statSync(result[1]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Custom Templates (Directory)
+result = shell.mktemp('-d', path.resolve('.', 'tmp/tmp.AXXX'), path.resolve('.', 'tmp/tmp.BXXX'));
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 2);
+assert.equal(result[0].slice(0, -3), path.resolve('.', 'tmp/tmp.A'));
+assert.equal(result[1].slice(0, -3), path.resolve('.', 'tmp/tmp.B'));
+assert.ok(fs.existsSync(result[0]));
+assert.ok(fs.statSync(result[0]).isDirectory());
+assert.ok(fs.existsSync(result[1]));
+assert.ok(fs.statSync(result[1]).isDirectory());
+if (isNotWindows) assert.equal(fs.statSync(result[0]).mode & parseInt('777', 8), parseInt('0600', 8));
+
+// Unsafe Mode
+result = shell.mktemp('-u');
+assert.equal(shell.error(), null);
+assert.equal(result.code, 0);
+assert.equal(result.length, 1);
```

Comment:
Could you also verify that this defaults to an appropriate temp directory, for both Unix and Windows?